plugin/backend/ipvariant.d

```diff
@@ -628,18 +629,39 @@ body {
     // recursive to handle nested namespaces.
     // the singleton ns must be the first code generate or the impl can't
     // use the instance.
-    static void eachNs(LookupT)(CppNamespace ns, Parameters params,
-            Generator.Modules modules, CppModule impl_singleton, LookupT lookup) {
+    @trusted static void eachNs(LookupT)(CppNamespace ns, Parameters params,
+            Generator.Modules modules, CppModule impl_singleton, LookupT lookup, string lastns) {
         
-        params.getSut;
-        //writeln("XML file " ~ params.getFiles.xml_interface);
-        auto currns =  ns.name;
-        auto inner = modules;
-        CppModule inner_impl_singleton;
+        string currns, currnsrp; 
+        if (ns.name != lastns) 
+        {
+            currns = lastns ~ "::" ~ ns.name;
+        }
+        else 
+        {
+            currns = ns.name;
+        }
 
-        if (currns != "std")
+        auto currns_spl = currns.split("::");
+        if (currns_spl[$-1] == "Requirer" ||  currns_spl[$-1] == "Provider")
+        {
+            currnsrp = join(currns_spl[0..$-1], "::");
+        }
+        else 
         {
+            currnsrp = currns;
+        }
 
+        SUTEnv sut = params.getSut.GetSUTFromNamespace(currnsrp);
+        //writeln("XML file " ~ params.getFiles.xml_interface)
+        auto inner = modules;
+        CppModule inner_impl_singleton;
+        writeln(currns);
+        writeln(sut.ToString);
+        writeln(sut.valid);
+        if (sut.valid)
+        {
+            
```

Comment:
tip: ns.resideInNs() or ns.fullyQualifiedName()
I think that is basically what a large part of this logic is doing.